bl31/bl31.mk

```diff
@@ -12,6 +12,7 @@ BL31_SOURCES		+=	bl31/bl31_main.c				\
 				bl31/aarch64/runtime_exceptions.S		\
 				bl31/aarch64/crash_reporting.S			\
 				bl31/bl31_context_mgmt.c			\
+				bl31/plat_params.c				\
```

Comment:
Regarding the "official dynamic configuration mechanism", don't hold your breath on that. I intend to discuss the design at the next Linaro Connect conference in September, with a view to starting the implementation shortly after. So we're looking at towards the end of this year. In terms of tracking, we'll either use https://github.com/ARM-software/tf-issues/issues/454 or create a new (more specific) issue.

In terms of mechanism, we're still looking at using the device tree format, but we'll try not to lock all platforms into using this if it doesn't meet your needs. I can see that console config while the MMU is disabled is particularly problematic.